components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -1249,6 +1250,74 @@ void BraveWalletService::MigrateUserAssetsAddIsSpam(PrefService* prefs) {
   prefs->SetBoolean(kBraveWalletUserAssetsAddIsSpamMigrated, true);
 }
 
+bool ShouldMigrateRemovedPreloadedNetwork(PrefService* prefs,
+                                          mojom::CoinType coin,
+                                          const std::string& chain_id) {
+  const auto* custom_networks = prefs->GetDict(kBraveWalletCustomNetworks)
+                                    .FindList(GetPrefKeyForCoinType(coin));
+  if (custom_networks) {
+    for (const auto& network_value : *custom_networks) {
+      const auto* chain_id_each = network_value.GetDict().FindString("chainId");
+      if (!chain_id_each) {
+        continue;
+      }
+
+      if (base::ToLowerASCII(*chain_id_each) == chain_id) {
+        return false;
+      }
+    }
+  }
```

Comment:
nit: there is `CustomChainExists`

Reply:
https://github.com/brave/brave-core/pull/19574/commits/74afc0c520a943998b558e937568cc63aa3ea14f